### ISSUE TITLE
Definition of parallel and distributed training on multiple GPUs

### ISSUE THREAD

**Issue by njellinas** [NVIDIA/apex#744]
Hello,

(Sorry if this is a wrong place to post this, but since I have not found any proper explanation in other packages it will be a very simple answer for anyone who knows what I am talking about.)

PyTorch DataParallel splits the batch size according to how many GPUs I have on a single machine. This does not reduce the training time but allows for larger batch size.

So my question is: Can I run distributed training on the same machine on multiple GPUs? I.e. every GPU gets the same batch size as the original but runs the model in parallel, so that I get multiple gradient updates on a single step, i.e. faster training?
If this is feasible, does apex support it?

**Comment by malcolmyanguci** [comment]
Yes I think Apex support it, but you have to use apex.parallel.DistributedDataParallel after setting up the multi-process function. For example, `mp.spawn(train, nprocs=num_proccess, args=(opt,model,f_loss,optimizer,scheduler,train_dataset,),join=True)` For each GPU you can manually assign the batch. It totally depends on you. How many process on one single gpu and how many batches each process and how many batches on each gpu...etc.
take a look on `https://github.com/NVIDIA/apex/tree/master/examples/imagenet`
